apm-sniffer/bootstrap-plugins/jdk-http-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/http/HttpClientNewInstanceInterceptor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.jdk.http;
+
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.StaticMethodsAroundInterceptor;
+
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+
+/**
+ * @author lican
+ */
+public class HttpClientNewInstanceInterceptor implements StaticMethodsAroundInterceptor {
+
+
+    @Override
+    public void beforeMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes, MethodInterceptResult result) {
+
+    }
+
+    @Override
+    public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes, Object ret) {
+        if (ret instanceof EnhancedInstance) {
+            if (allArguments.length >= 5 && allArguments[4] instanceof HttpURLConnection) {
+                //allArguments[4] is HttpURLConnection,for HttpClient
+                ((EnhancedInstance) ret).setSkyWalkingDynamicField(allArguments[4]);
+            } else if (allArguments.length >= 7 && allArguments[6] instanceof HttpURLConnection) {
+                //for HttpsClient
+                ((EnhancedInstance) ret).setSkyWalkingDynamicField(allArguments[6]);
+            }
```

Comment:
This should be placed in instrumentation definition if possible. this check costs unnecessary performance, right?

Reply:
Yes

Reply:
Yes ,I am updating now